x-pack/plugins/rule_registry/server/search_strategy/search_strategy.ts

```diff
@@ -191,7 +223,15 @@ export const ruleRegistrySearchStrategyProvider = (
             );
           }
 
-          throw err;
+          if (Boom.isBoom(err)) {
```

Comment:
Search strategy was always thrown a `500` error without respecting the error codes, like 403, throwing by the alerts authorization. I tried to fix that.